op-node/rollup/derive/channel_compressor.go

```diff
@@ -0,0 +1,106 @@
+package derive
+
+import (
+	"bytes"
+	"compress/zlib"
+	"fmt"
+	"io"
+
+	"github.com/andybalholm/brotli"
+)
+
+const (
+	ChannelVersionBrotli byte = 0x01
+)
+
+type ChannelCompressor interface {
+	Write([]byte) (int, error)
+	Flush() error
+	Close() error
+	Reset()
+	Len() int
+	Read([]byte) (int, error)
+	GetCompressed() *bytes.Buffer
+}
+
+type CompressorWriter interface {
+	Write([]byte) (int, error)
+	Flush() error
+	Close() error
+	Reset(io.Writer)
+}
+
+type BaseChannelCompressor struct {
+	compressed *bytes.Buffer
+	writer     CompressorWriter
+}
+
+func (bcc *BaseChannelCompressor) Write(data []byte) (int, error) {
+	return bcc.writer.Write(data)
+}
+
+func (bcc *BaseChannelCompressor) Flush() error {
+	return bcc.writer.Flush()
+}
+
+func (bcc *BaseChannelCompressor) Close() error {
+	return bcc.writer.Close()
+}
```

Comment:
does this work?
```suggestion
type BaseChannelCompressor struct {
	compressed *bytes.Buffer
        CompressorWriter
}
```
then in `Reset()` you can access `zc.CompressorWriter.Reset()`